### PR TITLE
Add icons to Windows and MacOS bundles

### PR DESCRIPTION
**Pull request by Lgt2x**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [x] Build and Dependency changes
- [ ] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [ ] Other changes

### Description

For Windows: set .rc file pointing .ico file
For MacOS: create .icns icone file and reference it as resource in .plist. Make MacOS executable into a .app

### CODE REVIEW

**Comment by Lgt2x** [comment]
(ignore wrong branch name, this was originally an installer draft, which will be the next PR)

**Comment by winterheart** [comment]
I think it's better to use `SDL_SetWindowIcon()` which would be cross-platform approach.

**Comment by Lgt2x** [comment]
> I think it's better to use `SDL_SetWindowIcon()` which would be cross-platform approach.

This is not the icon for runtime, but the icon that appears in the file manager and desktop for the app or exe

**Comment by pzychotic** [comment]
While we are adding a .rc file anyway, can we add the VS_VERSION_INFO block also? So the properties dialog of the Descent3.exe will show proper version and other properties.
Might create the version numbers through `configure_file` from CMake as we already do with the manifest.

```
#define APSTUDIO_READONLY_SYMBOLS
#include "winres.h"
#undef APSTUDIO_READONLY_SYMBOLS

/////////////////////////////////////////////////////////////////////////////
//
// Version
//

VS_VERSION_INFO VERSIONINFO
 FILEVERSION 1,5,0,0
 PRODUCTVERSION 1,5,0,0
 FILEFLAGSMASK 0x3fL
#ifdef _DEBUG
 FILEFLAGS 0x1L
#else
 FILEFLAGS 0x0L
#endif
 FILEOS 0x40004L
 FILETYPE 0x1L
 FILESUBTYPE 0x0L
BEGIN
    BLOCK "StringFileInfo"
    BEGIN
        BLOCK "040004b0"
        BEGIN
            VALUE "CompanyName", "TODO: <Company name>"
            VALUE "FileDescription", "TODO: <File description>"
            VALUE "FileVersion", "1.5.0.0"
            VALUE "InternalName", "Descent3.exe"
            VALUE "LegalCopyright", "Copyright (C) 2024"
            VALUE "OriginalFilename", "Descent3.exe"
            VALUE "ProductName", "Descent 3"
            VALUE "ProductVersion", "1.5.0.0"
        END
    END
    BLOCK "VarFileInfo"
    BEGIN
        VALUE "Translation", 0x400, 1200
    END
END
```

**Comment by Lgt2x** [comment]
Thanks for the suggestion, I added that

File: Descent3/Descent3.rc.in [hunk 1]
```diff
@@ -0,0 +1,38 @@
+#define APSTUDIO_READONLY_SYMBOLS
+#include "winres.h"
+#undef APSTUDIO_READONLY_SYMBOLS
+
+IDI_ICON1 ICON DISCARDABLE "Descent3.ico"
+
+VS_VERSION_INFO VERSIONINFO
+ FILEVERSION 1,5,0,0
+ PRODUCTVERSION 1,5,0,0
```
Comment (pzychotic) [inline Descent3/Descent3.rc.in:9]: These two version entries might need the @ replacement, too.

Comment (Lgt2x) [reply Descent3/Descent3.rc.in]: yep, fixed

File: Descent3/Descent3.rc.in [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ FILEOS 0x40004L
+ FILETYPE 0x1L
+ FILESUBTYPE 0x0L
+BEGIN
+    BLOCK "StringFileInfo"
+    BEGIN
+        BLOCK "040004b0"
+        BEGIN
+            VALUE "CompanyName", "Descent Developpers"
+            VALUE "FileDescription", "Descent 3 Open Source game"
```
Comment (winterheart) [inline Descent3/Descent3.rc.in:25]: Can you please add https://github.com/DescentDevelopers/Descent3 URL here?

File: Descent3/Descent3.rc.in [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+            VALUE "FileVersion", "@PROJECT_VERSION_MAJOR@.@PROJECT_VERSION_MINOR@.@PROJECT_VERSION_PATCH@.0"
+            VALUE "InternalName", "Descent3.exe"
+            VALUE "LegalCopyright", "Copyright (C) 2024"
```
Comment (winterheart) [inline Descent3/Descent3.rc.in:28]: We need add our standard string from header

```
Copyright (C) 2024 Descent Developers
```

**Review by winterheart — CHANGES_REQUESTED** [review]
Generally compiles on Windows as expected. I'd request some changes though.

**Comment by Lgt2x** [comment]
Review comments fixed

**Pull request merged**